### DISCUSSION TITLE
Does BeeWare Support Hot Reload?

### DISCUSSION THREAD

**Discussion by jorpyy** [beeware/beeware#441] [Q&A] [answered]
Does BeeWare support hot reload?

**Comment by freakboy3742** [comment, marked as answer]
No - there's no hot reload capabilities in BeeWare tools at present.

If they existed, it would likely be part of Toga; but hot reload is difficult to perform with a mobile-based app, and especially a Python one. Hot reload in a web browser is largely possible because the web itself is a stateless platform, and each page you view is an independent (and complete) application state. However, a mobile app isn't - the app itself is a running executable, so you need to be able to resume the complete application state on reload.

That doesn't mean it's *impossible* - just a lot more difficult.
